bcftbx/cmdparse.py

```diff
@@ -157,33 +162,36 @@ def parse_args(self,argv=None):
         except KeyError:
             # No parser
             self.error("Usage: %s COMMAND [options] [args...]\n\n"
-                       "%s: error: no such command: %s" % 
-                       (self._name,self._name,cmd))
+                       "%s: error: no such command: %s" %
+                       (self._name, self._name, cmd))
+
         # Parse the remaining arguments and return
-        options,arguments = p.parse_args(argv[1:])
-        return (cmd,options,arguments)
+        if isinstance(p, argparse.ArgumentParser):
+            options, arguments = p.parse_known_args(argv[1:])
```

Comment:
I'm not happy with the change of using `parse_known_args` instead of `parse_args`, as I feel like this could mask errors in parsing the command line when using `argparse` as the subparser, and is not consistent with the behaviour when using `optparse`.

  My preference would be to return the `(cmd,options,arguments)` tuple if `optparse` is the subparser, and just `(cmd,arguments)` if `argparse` is being used, e.g.:

      def parse_args(self,argv=None):
          ...
          # Parse the remaining arguments and return
          if isinstance(p,argparse.ArgumentParser):
              arguments = p.parse_args(argv[1:])
              return (cmd,arguments)
          else:
              options, arguments = p.parse_args(argv[1:])
              return (cmd, options, arguments)